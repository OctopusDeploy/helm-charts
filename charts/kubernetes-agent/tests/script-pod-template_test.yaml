suite: "Script Pod Template"
templates:
  - templates/script-pod-template.yaml
tests:
  - it: should not render when disabled
    set:
      scriptPods:
        podTemplate:
          enabled: false
    asserts:
      - hasDocuments:
          count: 0

  - it: should partially render successfully
    set:
      scriptPods:
        podTemplate:
          enabled: true
          scriptContainerSpec:
            resources:
              limits:
                memory: "256Mi"
                cpu: "500m"
    asserts:
      - matchSnapshot: {}

  - it: "should match snapshot"
    set:
      scriptPods:
        podTemplate:
          enabled: true
          podSpec:
            nodeSelector:
              disktype: ssd
          scriptContainerSpec:
            resources:
              limits:
                memory: "256Mi"
                cpu: "500m"
          watchdogContainerSpec:
            resources:
              limits:
                memory: "256Mi"
                cpu: "500m"
    asserts:
      - matchSnapshot: {}