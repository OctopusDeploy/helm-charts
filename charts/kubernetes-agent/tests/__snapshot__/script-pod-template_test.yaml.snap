should match snapshot:
  1: |
    apiVersion: agent.octopus.com/v1beta1
    kind: ScriptPodTemplate
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: octopus-agent
        app.kubernetes.io/version: 8.3.3193
        helm.sh/chart: kubernetes-agent-2.24.1
      name: octopus-agent
      namespace: NAMESPACE
    spec:
      podMetadata:
        annotations: {}
        labels: {}
      podSpec:
        nodeSelector:
          disktype: ssd
      scriptContainerSpec:
        resources:
          limits:
            cpu: 500m
            memory: 256Mi
      watchdogContainerSpec:
        resources:
          limits:
            cpu: 500m
            memory: 256Mi
should partially render successfully:
  1: |
    apiVersion: agent.octopus.com/v1beta1
    kind: ScriptPodTemplate
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: octopus-agent
        app.kubernetes.io/version: 8.3.3193
        helm.sh/chart: kubernetes-agent-2.24.1
      name: octopus-agent
      namespace: NAMESPACE
    spec:
      podMetadata:
        annotations: {}
        labels: {}
      podSpec: {}
      scriptContainerSpec:
        resources:
          limits:
            cpu: 500m
            memory: 256Mi
      watchdogContainerSpec: {}
